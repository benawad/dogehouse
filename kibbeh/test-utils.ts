import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
