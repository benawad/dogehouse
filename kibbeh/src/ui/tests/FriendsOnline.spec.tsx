import React from "react";

import { render } from "../../../test-utils";
import {
  FriendsOnline,
  FriendsOnlineProps,
  FriendOnlineType,
} from "../FriendsOnline";

jest.useFakeTimers();

const friendOnline: FriendOnlineType = {
  avatar: "",
  isOnline: true,
  username: "Some name",
  activeRoom: {
    name: "Some room",
    link: "",
  },
};

const showMoreAction = jest.fn();

let defaultProps: FriendsOnlineProps = {
  onlineFriendCount: 1,
  onlineFriendList: [friendOnline],
  showMoreAction,
};

describe("FriendsOnline", () => {
  beforeEach(() => {
    defaultProps = {
      ...defaultProps,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const { getByTestId, getByText } = render(
      <FriendsOnline {...defaultProps} />
    );
    const component = getByTestId("friends-online");

    expect(component).toBeVisible();
  });

  it("should render You have 0 friends online right now if online friends list length is 0", () => {
    const props = {
      ...defaultProps,
      onlineFriendList: [],
    };
    const { getByTestId } = render(<FriendsOnline {...props} />);
    const placeholder = getByTestId("placeholder");

    expect(placeholder).toBeVisible();
  });

  it("should call showMoreAction if Show more button clicked", () => {
    const { getByTestId } = render(<FriendsOnline {...defaultProps} />);
    const showMoreBtn = getByTestId("show-more-btn");
    showMoreBtn.click();

    expect(showMoreAction).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<FriendsOnline {...defaultProps} />);
    const component = getByTestId("friends-online");

    expect(component).toMatchSnapshot();
  });
});
