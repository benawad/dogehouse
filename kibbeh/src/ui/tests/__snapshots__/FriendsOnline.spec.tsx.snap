// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FriendsOnline should match snapshot 1`] = `
<div
  class="pb-5 w-full flex flex-col"
  data-testid="friends-online"
>
  <h4
    class="text-primary-100"
  >
    People
  </h4>
  <h6
    class="text-primary-300 mt-3 text-sm font-bold"
  >
    ONLINE (
    1
    )
  </h6>
  <div
    class="flex flex-col mt-3"
  >
    <div
      class="py-3 w-full"
    >
      <div
        class="relative inline-block "
        data-testid="single-user-avatar"
        style="width: 40px; height: 40px;"
      >
        <img
          alt="avatar"
          class="rounded-full w-full h-full"
          src=""
        />
        <span
          class="rounded-full absolute box-content bg-accent border-primary-800"
          data-testid="online-indictor"
          style="width: 8px; height: 8px; right: 2px; bottom: -2px; border-width: 2px;"
        />
      </div>
      <div
        class="ml-3 flex flex-col"
      >
        <h5
          class="text-primary-100 font-bold"
        >
          Some name
        </h5>
        <a
          class="text-primary-300 border-b"
        >
          Some room
        </a>
      </div>
    </div>
    <div
      class="underline text-primary-300 font-bold mt-4 cursor-pointer"
      data-testid="show-more-btn"
    >
      Show more
    </div>
  </div>
</div>
`;
